site/app/models/gradeable/AutoGradedVersion.php

```diff
@@ -332,6 +352,65 @@ public function getTotalPercent($clamp = false) {
         return $result;
     }
 
+    /**
+     * Gets the number of days late this version is
+     * @return int result clamped to be >= 0
+     */
+    public function getDaysLate() {
+        return max(0, DateUtils::calculateDayDiff(
+            (clone $this->getGradedGradeable()->getGradeable()->getSubmissionDueDate())
+                ->add(new \DateInterval("PT5M")), $this->submission_time));
```

Comment:
Grace period? Should probably be removed.

Reply:
I agree